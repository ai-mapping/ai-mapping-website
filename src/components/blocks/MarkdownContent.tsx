import React from 'react';
import Section from '@/components/layout/Section';
import ReactMarkdown from 'react-markdown';
import { cn } from '@/lib/utils';

// Main props interface
export type MarkdownContentProps = {
  content: string;
  className?: string;
  background?: 'surface' | 'background' | 'primary' | 'secondary' | 'accent' | 'none';
  paddingY?: 'sm' | 'md' | 'lg' | 'none';
  containerWidth?: 'full' | 'container' | 'narrow' | 'medium';
  withContainer?: boolean;
};

// Default props
export const defaultMarkdownContentProps: MarkdownContentProps = {
  content: '',
  background: 'background',
  paddingY: 'lg',
  containerWidth: 'medium',
  withContainer: true,
};

const MarkdownContent = ({
  content = defaultMarkdownContentProps.content,
  className,
  background = defaultMarkdownContentProps.background,
  paddingY = defaultMarkdownContentProps.paddingY,
  containerWidth = defaultMarkdownContentProps.containerWidth,
  withContainer = defaultMarkdownContentProps.withContainer,
}: Partial<MarkdownContentProps>) => {
  const markdownContent = (
    <div className="prose prose-xl max-w-none text-white">
      <ReactMarkdown
        components={{
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          h1: ({ node, ...props }) => (
            <h1
              className="text-3xl md:text-4xl font-bold mb-6 text-primary"
              {...props}
            />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          h2: ({ node, ...props }) => (
            <h2
              className="text-2xl md:text-3xl font-semibold mb-5 mt-8 text-primary"
              {...props}
            />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          h3: ({ node, ...props }) => (
            <h3
              className="text-xl md:text-2xl font-semibold mb-4 mt-6 text-primary"
              {...props}
            />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          p: ({ node, ...props }) => (
            <p className="text-lg mb-5 text-neutral leading-relaxed" {...props} />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          strong: ({ node, ...props }) => (
            <strong className="font-semibold text-white" {...props} />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          ul: ({ node, ...props }) => (
            <ul className="space-y-4 my-6 text-lg" {...props} />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          li: ({ node, ...props }) => (
            <li className="flex items-start">
              <div className="h-6 w-6 bg-primary rounded-full flex items-center justify-center mr-3 mt-0.5 flex-shrink-0">
                <svg xmlns="http://www.w3.org/2000/svg" className="h-3 w-3 text-black" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="3" d="M12 5v14M5 12h14" />
                </svg>
              </div>
              <span className="text-neutral" {...props} />
            </li>
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          blockquote: ({ node, ...props }) => (
            <blockquote
              className="border-l-4 border-primary pl-6 italic bg-neutral-light p-5 rounded-r my-6 text-lg"
              {...props}
            />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          a: ({ node, ...props }) => (
            <a className="text-accent hover:text-accent-2 transition-colors duration-300 font-medium" {...props} />
          ),
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          code: ({ node, inline, ...props }) => (
            inline
              ? <code className="bg-neutral-light px-1.5 py-0.5 rounded text-accent font-mono" {...props} />
              : <code className="block p-4 bg-neutral-light rounded-lg overflow-x-auto my-6 font-mono text-sm" {...props} />
          ),
        }}
      >
        {content}
      </ReactMarkdown>
    </div>
  );

  return (
    <Section
      background={background}
      paddingY={paddingY}
      containerWidth={containerWidth}
    >
      {withContainer ? (
        <div className={cn('bg-surface p-6 rounded-lg shadow-lg border border-neutral-dark', className)}>
          {markdownContent}
        </div>
      ) : (
        <div className={className}>{markdownContent}</div>
      )}
    </Section>
  );
};

// For backwards compatibility
export type RegionalProps = MarkdownContentProps;
export const defaultRegionalProps = {
  ...defaultMarkdownContentProps,
  content: `## Доставка запчастей Caterpillar в Москве и области

Мы обеспечиваем быструю доставку запчастей Caterpillar и KOMATSU по Москве и Московской области. При наличии запчастей на нашем складе, возможна доставка в день заказа или на следующий день.

**Для компаний из Москвы и области мы предлагаем дополнительные преимущества:**

- Бесплатная курьерская доставка при заказе от 50 000 рублей
- Возможность самовывоза со склада в Москве
- Выезд технического специалиста для консультации и диагностики (по договоренности)
- Возможность оперативной доставки срочных заказов

Наш офис находится в Москве, что позволяет нам оперативно решать все вопросы, связанные с поставками запчастей для клиентов из столичного региона.`,
};

export default MarkdownContent;
