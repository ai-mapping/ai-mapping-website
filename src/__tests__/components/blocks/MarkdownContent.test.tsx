import React from 'react';
import { render } from '@testing-library/react';
import MarkdownContent, { defaultMarkdownContentProps } from '@/components/blocks/MarkdownContent';

// Mock the react-markdown component
jest.mock('react-markdown', () => {
  return ({ children }: { children: string }) => <div data-testid="markdown">{children}</div>;
});

describe('MarkdownContent Component', () => {
  it('renders with default props correctly', () => {
    const { container, getByTestId } = render(<MarkdownContent content="Test content" />);
    
    // Check if markdown content is passed to the component
    const markdownElement = getByTestId('markdown');
    expect(markdownElement.textContent).toBe("Test content");
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom content correctly', () => {
    const customProps = {
      content: '### Custom Markdown Content\n\nThis is a test.',
      background: 'white' as const,
      paddingY: 'sm' as const
    };
    
    const { container, getByTestId } = render(<MarkdownContent {...customProps} />);
    
    // Check if custom markdown content is passed to the component
    const markdownElement = getByTestId('markdown');
    expect(markdownElement.textContent).toBe(customProps.content);
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});