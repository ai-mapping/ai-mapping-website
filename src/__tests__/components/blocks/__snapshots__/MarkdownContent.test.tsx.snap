// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownContent Component renders with custom content correctly 1`] = `
<div>
  <section
    class="py-6"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-4xl"
    >
      <div
        class="bg-surface p-6 rounded-lg shadow-lg border border-neutral-dark"
      >
        <div
          class="prose prose-lg max-w-none text-white"
        >
          <div
            data-testid="markdown"
          >
            ### Custom Markdown Content

This is a test.
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`MarkdownContent Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-4xl"
    >
      <div
        class="bg-surface p-6 rounded-lg shadow-lg border border-neutral-dark"
      >
        <div
          class="prose prose-lg max-w-none text-white"
        >
          <div
            data-testid="markdown"
          >
            Test content
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
